cms/bundles/views.py

```diff
@@ -0,0 +1,104 @@
+from typing import TYPE_CHECKING, Any
+
+from django.core.exceptions import PermissionDenied
+from django.http import Http404
+from django.shortcuts import get_object_or_404, redirect
+from django.urls import reverse
+from django.utils.http import url_has_allowed_host_and_scheme
+from django.utils.translation import gettext as _
+from django.views.generic import FormView
+from wagtail.admin import messages
+from wagtail.models import Page
+from wagtail.permission_policies import ModelPermissionPolicy
+
+from .admin_forms import AddToBundleForm
+from .models import Bundle, BundledPageMixin, BundlePage
+
+if TYPE_CHECKING:
+    from django.http import HttpRequest, HttpResponseBase, HttpResponseRedirect
+
+
+class AddToBundleView(FormView):
+    form_class = AddToBundleForm
+    template_name = "bundles/wagtailadmin/add_to_bundle.html"
+
+    page_to_add: Page = None
+    goto_next: str | None = None
+
+    def dispatch(self, request: "HttpRequest", *args: Any, **kwargs: Any) -> "HttpResponseBase":
+        self.page_to_add = get_object_or_404(
+            Page.objects.specific().defer_streamfields(), id=self.kwargs["page_to_add_id"]
+        )
+
+        if not isinstance(self.page_to_add, BundledPageMixin):
+            raise Http404(_("Cannot add this page type to a bundle"))
+
+        page_perms = self.page_to_add.permissions_for_user(request.user)  # type: ignore[attr-defined]
+        # TODO: add the relevant permission checks
+        if not (page_perms.can_edit() or page_perms.can_publish()):
+            raise PermissionDenied
+
+        permission_policy = ModelPermissionPolicy(Bundle)
+        if not permission_policy.user_has_permission(request.user, "change"):
+            raise PermissionDenied
+
+        self.goto_next = None
+        redirect_to = request.GET.get("next", "")
+        if url_has_allowed_host_and_scheme(url=redirect_to, allowed_hosts={self.request.get_host()}):
+            self.goto_next = redirect_to
+
+        if self.page_to_add.in_active_bundle:
+            bundles = ", ".join(list(self.page_to_add.active_bundles.values_list("name", flat=True)))
```

Comment:
Suggestion: `django.utils.text.get_text_list` should join these quite nicely, and stick an "and" for the final one.

Reply:
Nice one! forgot about it